js/renovation/scheduler/workspaces/day/date_table/layout.tsx

```diff
@@ -5,47 +5,58 @@ import { DateTableRow } from '../../base/date_table/row';
 import { DayDateTableCell as Cell } from './cell';
 import { GroupedViewData, ViewCellData } from '../../types';
 import { getKeyByDateAndGroup } from '../../utils';
+import { Table } from '../../base/table';
+import { VirtualTable } from '../../base/virtual_table';
 
-export const viewFunction = (viewModel: DayDateTableLayout) => (
-  <table
-    className={`${viewModel.props.className} dx-scheduler-date-table`}
-    // eslint-disable-next-line react/jsx-props-no-spreading
-    {...viewModel.restAttributes}
-  >
-    <tbody>
-      {viewModel.props.viewCellsData!
-        .groupedData.map(({ dateTable }) => dateTable.map((cellsRow) => (
-          <DateTableRow
-            key={getKeyByDateAndGroup(cellsRow[0].startDate, cellsRow[0].groups)}
-          >
-            {cellsRow.map(({
-              startDate,
-              endDate,
-              groups,
-            }: ViewCellData) => (
-              <Cell
-                startDate={startDate}
-                endDate={endDate}
-                groups={groups}
-                key={getKeyByDateAndGroup(startDate, groups)}
-              />
-            ))}
-          </DateTableRow>
-        )))}
-    </tbody>
-  </table>
-);
+export const viewFunction = (viewModel: DayDateTableLayout): object => {
+  const DateTable = viewModel.props.isVirtual ? VirtualTable : Table;
+
+  return (
+    <DateTable
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...viewModel.restAttributes}
+      className={`dx-scheduler-date-table ${viewModel.props.className}`}
+    >
+      {
+          viewModel.props.viewData!
+            .groupedData.map(({ dateTable }) => dateTable.map((cellsRow) => (
+              <DateTableRow
+                key={getKeyByDateAndGroup(cellsRow[0].startDate, cellsRow[0].groups)}
+              >
+                {cellsRow.map(({
+                  startDate,
+                  endDate,
+                  groups,
+                }: ViewCellData) => (
+                  <Cell
+                    startDate={startDate}
+                    endDate={endDate}
+                    groups={groups}
+                    key={getKeyByDateAndGroup(startDate, groups)}
+                  />
+                ))}
+              </DateTableRow>
+            )))
+      }
+    </DateTable>
+  );
+};
 
 @ComponentBindings()
 export class DayDateTableLayoutProps {
-  @OneWay() viewCellsData?: GroupedViewData;
+  @OneWay() viewData?: GroupedViewData;
 
   @OneWay() className?: string;
+
+  @OneWay() isVirtual?: boolean;
 }
 
 @Component({
   defaultOptionRules: null,
   view: viewFunction,
 })
 export class DayDateTableLayout extends JSXComponent(DayDateTableLayoutProps) {
+  get DateTable(): typeof VirtualTable | typeof Table {
+    return this.props.isVirtual ? VirtualTable : Table;
+  }
```

Comment:
We should either use it or delete it. if we use it, this getter should be tested

Reply:
I've already tested it in the tests of the render:
https://github.com/DevExpress/DevExtreme/pull/13449/files/59efef5eafeb3428902e6c7ed2166e28bbddb9be#diff-1b1e8943bf4b91173fa173151f84a343R40-R52
Because checking of this logic needs rendering of the components.

Reply:
Getters should not be tested by rendering a component. And in your case it is not tested in render method. View function contains an exactly the same [piece of code](https://github.com/DevExpress/DevExtreme/pull/13449/files#diff-a4cd4307ff5fb4a0544f0a3bfd223077R12). So, it should be removed either from viewFunction or from here. If it is removed from here, there is no need for a new test. If it is removed from viewFunction though, this getter should be tested (without rendering a component).

Reply:
Ok than. If it is so principle...
I will replace this getter with a piece of logic in the view function.

Reply:
This getter is already redundant. I'll just remove it.